Code/GraphMol/MolDraw2D/DrawMol.cpp

```diff
@@ -1411,16 +1411,31 @@ void DrawMol::extractLegend() {
   double fsize = textDrawer_.fontSize();
   double relFontScale = drawOptions_.legendFontSize / fsize;
   double total_width, total_height;
-  double adjLegHt = legendHeight_ - 0.5 * drawOptions_.padding * height_;
   calc_legend_height(legend_bits, relFontScale, total_width, total_height);
-  if (total_height > adjLegHt) {
-    relFontScale *= double(adjLegHt) / total_height;
-    calc_legend_height(legend_bits, relFontScale, total_width, total_height);
+  if (total_width >= width_) {
+    if (!flexiCanvasX_) {
+      relFontScale *= double(width_) / total_width;
+      calc_legend_height(legend_bits, relFontScale, total_width, total_height);
+    } else {
+      width_ = total_width;
+      width_ += width_ * drawOptions_.padding;
+    }
   }
-  if (total_width > width_) {
-    relFontScale *= double(width_) / total_width;
-    calc_legend_height(legend_bits, relFontScale, total_width, total_height);
+
+  if (!flexiCanvasY_) {
+    auto adjLegHt = height_ * drawOptions_.legendFraction;
+    // subtract off space for the padding.
+    adjLegHt -= 0.5 * drawOptions_.padding * height_;
+    if (total_height > adjLegHt) {
+      relFontScale *= double(adjLegHt) / total_height;
+      calc_legend_height(legend_bits, relFontScale, total_width, total_height);
+    }
+  } else {
+    height_ += total_height;
+    // REVIEW: is this factoring in the padding for the drawing bit twice?
```

Comment:
Yes, I think it is.  If it's a full flexicanvas the padding is added at DrawMol.cc line 800.  If it's a flexiCanvasY_ it's added in DrawMol::findExtremes().